### PR TITLE
Download the cx-server from github.com

### PR DESCRIPTION
**Pull request by florian-richter**
## Proposed Changes

This changes the behavior of `sap-cloud-sdk add-cx-server` to get the cx-server and server.cfg from the master branch of the respective github repo instead of copying files locally.

This has the benefit that the files will always be the newest version, but it now requires internet connectivity to run (which can be assumed as the scripts need internet connectivity to run anyways).

## Type of Changes

- [ ] Bug Fix
- [x] New Feature
- [ ] Documentation Update

## Breaking Changes

In general avoid breaking changes, but if you think it is necessary give your reasoning in the _Further Comments_ section.

- [x] I have not introduced breaking changes

### CODE REVIEW

File: src/utils/templates.ts [hunk 1]
```diff
@@ -55,25 +56,59 @@ export async function findConflicts(files: CopyDescriptor[], force: boolean, std
...
+        });
+      })
+      .on('error', e => {
+        reject(e);
+      });
+  });
+}
+
+function copyLocal(sourcePath: string, targetFolder: string, fileName: string, options: { [key: string]: any }) {
+  return new Promise((resolve, reject) => {
```
Comment (mr-flannery) [inline src/utils/templates.ts]: nitpick: you could save one level of indentation across the whole function by returning `Promise.resolve()` or `Promise.reject(e)`, respectively. Personally I find this easier to read.

File: src/commands/init.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export default class Init extends Command {
       cli.action.stop();
 
       cli.action.start('Creating files');
-      copyFiles(files, options, this.error);
+      copyFiles(files, options).catch(e => this.error(e, { exit: 2 }));
```
Comment (mr-flannery) [inline src/commands/init.ts]: Shouldn't there be an `await` here now that the function is `async`?

File: src/commands/add-cx-server.ts [hunk 3]
```diff
@@ -29,16 +29,26 @@ export default class AddCxServer extends Command {
...
+          sourcePath: new URL('cx-server', githubPrefix),
+          targetFolder: path.resolve(flags.projectDir, 'cx-server'),
+          fileName: path.resolve(flags.projectDir, 'cx-server', 'cx-server')
+        },
+        {
+          sourcePath: new URL('server.cfg', githubPrefix),
+          targetFolder: path.resolve(flags.projectDir, 'cx-server'),
+          fileName: path.resolve(flags.projectDir, 'cx-server', 'server.cfg')
+        }
+      ];
```
Comment (mr-flannery) [inline src/commands/add-cx-server.ts]: what happened to the `.bat` file? Shouldn't we keep Windows support?

Comment (mr-flannery) [reply src/commands/add-cx-server.ts]: Also what happened to the concern about the number of relevant files potentially changing? No proper/sane way to implement it?

File: src/utils/templates.ts (hunk 1 above)
Comment (florian-richter) [reply src/utils/templates.ts]: Can you be more explicit? I tried to do it that way but I couldn't figure out how. But maybe I am missing something obvious...

File: src/commands/init.ts (hunk 2 above)
Comment (florian-richter) [reply src/commands/init.ts]: Good catch!

File: src/commands/add-cx-server.ts (hunk 3 above)
Comment (florian-richter) [reply src/commands/add-cx-server.ts]: So this is the result from a discussion with the pipeline team. They said that they consider only those two files essential and also consider the naming and folder structure as stable. So this seemed like the easiest way.

There are downsides, like the risk of breaking this command when the file names change. But I feel any other solution would likely be over-engineered or slow.

Henning also pointed out that we should keep the `.bat` while the pipeline team seemed to dismiss it. I will add it back in if the command is executed on a Windows machine.

**Pull request merged**